iib/workers/tasks/build.py

```diff
@@ -755,25 +755,55 @@ def _adjust_operator_manifests(manifests_path):
     For any container image pull spec found in the Operator CSV files, replace floating
     tags with pinned digests, e.g. `image:latest` becomes `image@sha256:...`.
 
+    If spec.relatedImages is not set, it will be set with the pinned digests. If it is set but
+    there are also RELATED_IMAGE_* environment variables set, an exception will be raised.
+
     This method relies on the OperatorManifest class to properly identify and apply the
     modifications as needed.
 
     :param str manifests_path: the full path to the directory containing the operator manifests.
+    :raises IIBError: if the operator manifest has invalid entries
     """
     operator_manifest = OperatorManifest.from_directory(manifests_path)
     found_pullspecs = set()
+    operator_csvs = []
     for operator_csv in operator_manifest.files:
+        if operator_csv.has_related_images():
+            csv_file_name = os.path.basename(operator_csv.path)
+            if operator_csv.has_related_image_envs():
+                raise IIBError(
+                    f'The ClusterServiceVersion file {csv_file_name} has entries in '
```

Comment:
Just out of curiosity, can I ask why is that we raise this error?

Reply:
It was a policy decision to limit confusing behavior of setting it in multiple ways.